.github/scripts/compile.sh

```diff
@@ -42,13 +42,19 @@ tar -zxf qpid-proton.tar.gz -C qpid-proton-src --strip-components 1
 do_patch "patches/proton" qpid-proton-src
 
 cd proton_build
-cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_BINDINGS=python -DCMAKE_INSTALL_PREFIX=/usr -DSYSINSTALL_PYTHON=ON -DSSL_IMPL=openssl $WORKING/qpid-proton-src/ \
-    && make \
-    && make DESTDIR=$WORKING/proton_install install \
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DENABLE_LINKTIME_OPTIMIZATION=ON \
+  -DSSL_IMPL=openssl -DBUILD_STATIC_LIBS=ON -DBUILD_BINDINGS=python -DSYSINSTALL_PYTHON=ON \
```

Comment:
Static proton libs don't get compiled with LTO. I think this is a deliberate decision (by @astitcher, or maybe Alan, back then) and not an omission. The problem with LTO-ing static libraries is that you have to replace `ar` and `runlib` with `gcc-ar` or `gcc-runlib`, and there is something equivalent for clang... Or you can try to use CMake facilities if you have new-enough CMake (https://cmake.org/cmake/help/latest/variable/CMAKE_INTERPROCEDURAL_OPTIMIZATION.html#variable:CMAKE_INTERPROCEDURAL_OPTIMIZATION).

In any case, it is not exactly simple, then.

Reply:
@astitcher Do you maybe remember the details about ^^^? If the `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON` behaves better, would you be for enabling LTO that way?

Btw, new Fedora (and maybe RHEL 9) enable LTO by default when compiling RPM packages, e.g.

```
+ CFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection'
```

from https://download.copr.fedorainfracloud.org/results/jdanek/skupper-router/fedora-36-x86_64/03946420-skupper-router/builder-live.log.gz (This is Fedora 36)

Reply:
`CMAKE_INTERPROCEDURAL_OPTIMIZATION` does indeed behave better, the winning compilation options are

```
-DCMAKE_POLICY_DEFAULT_CMP0069=NEW -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON 
```